spec/plugin.spec.js

```diff
@@ -393,4 +393,85 @@ describe('ManifestPlugin', function() {
       });
     });
   });
+
+  describe('filter', function() {
+    it('should filter out non-initial chunks', function(done) {
+      webpackCompile({
+        context: __dirname,
+        entry: {
+          nameless: './fixtures/nameless.js'
+        },
+        output: {
+          filename: '[name].[hash].js'
+        }
+      }, {
+        manifestOptions: {
+          filter: function(file) {
+            return file.isInitial;
+          }
+        }
+      }, function(manifest, stats) {
+        expect(Object.keys(manifest).length).toEqual(1);
+        expect(manifest['nameless.js']).toEqual('nameless.'+ stats.hash +'.js');
+
+        done();
+      });
+    });
+  });
+
+  describe('map', function() {
+    it('should allow modifying files defails', function(done) {
+      webpackCompile({
+        context: __dirname,
+        entry: './fixtures/file.js',
+        output: {
+          filename: '[name].js'
+        }
+      }, {
+        manifestOptions: {
+          map: function(file, i) {
+            file.name = i.toString();
+            return file;
+          }
+        }
+      }, function(manifest, stats) {
+        expect(manifest).toEqual({
+          '0': 'main.js'
+        });
+
+        done();
+      });
+    });
+  });
+
+  describe('reduce', function() {
+    it('should generate custom manifest', function(done) {
+      webpackCompile({
+        context: __dirname,
+        entry: './fixtures/file.js',
+        output: {
+          filename: '[name].js'
+        }
+      }, {
+        manifestOptions: {
+          reduce: function (manifest, file) {
+            manifest[file.name] = {
+              file:file.path,
```

Comment:
We definitely need to setup eslint as some point

I personally have a preference for https://github.com/sindresorhus/xo , but really I don't mind ;)